python/tvm/driver/tvmc/autoscheduler.py

```diff
@@ -0,0 +1,212 @@
+        )
+    else:
+        logger.info("starting localhost tuning")
+        runner = auto_scheduler.LocalRunner(
+            number=args.number,
+            repeat=args.repeat,
+            timeout=args.timeout,
+            min_repeat_ms=min_repeat_ms,
+        )
+
+    # Create the autoscheduler tuning options
+    tuning_options = auto_scheduler.TuningOptions(
+        num_measure_trials=args.trials,
+        measure_callbacks=[auto_scheduler.RecordToFile(args.output)],
+        runner=runner,
+        builder="local",
+        early_stopping=args.early_stopping,
+    )
+
+    # Specify hardware parameters
+    hardware_params = HardwareParams(
+        args.num_cores, args.vector_unit_bytes, args.cache_line_bytes, None, None, None, None, None
+    )
+
+    # Extract the tasks from the model
+    tasks, weights = get_tuning_tasks(
+        mod, params, target, target_host, args.desired_layout, hardware_params
+    )
+
+    # Schedule the tasks (i.e., produce a schedule for each task)
+    schedule_tasks(
+        tasks,
+        weights,
+        tuning_options,
+        args.tuning_records,
+    )
+
+
+def get_tuning_tasks(
+    mod, params, target, target_host=None, alter_layout=None, hardware_params=None
+):
+    """Get the tuning tasks for a given relay module.
+
+    Parameters
+    ----------
+    mod : tvm.relay.Module
+        The relay module from which to extract tuning tasks.
+    params : dict
+        The params for the relay module.
+    target : tvm.target.Target
+        The compilation target.
+    target_host : str, optional
+        The compilation target for the host.
+    alter_layout : str, optional
+        The layout to convert the graph to. Note, the convert layout
+        pass doesn't currently guarantee the whole of the graph will
+        be converted to the chosen layout.
+
+    Returns
+    -------
+    tasks : list of autotvm.Tasks
+        list of tasks to be tuned
+    weights : List[int]
+        the weight (i.e. the number of appearance) of extracted tasks
+    """
+    if alter_layout:
+        mod = common.convert_graph_layout(mod, alter_layout)
+
+    # Extract the tasks
+    tasks, task_weights = auto_scheduler.extract_tasks(
+        mod["main"], params, target=target, target_host=target_host, hardware_params=hardware_params
+    )
+
+    return tasks, task_weights
+
+
+def schedule_tasks(
+    tasks,
+    task_weights,
+    tuning_options,
+    tuning_records=None,
+):
+    """Generate the schedules for the different tasks (i.e., subgraphs) contained in the module.
+    Store the schedules in a json file that will be used later by the compiler.
+
+    Parameters
+    ----------
+    tasks : list
+        A list of autotvm.Tasks to tune.
+    task_weights : list
+        The weight (i.e. the number of appearance) of extracted tasks
+    tuning_records : str, optional
+        The json file used to preload the autoscheduler
+    tuning_options:
+        The options of tuning
```

Comment:
Wrong order.

Reply:
File has been deleted but I applied this change